calculator_test.go

```diff
@@ -2,23 +2,295 @@ package calculator_test
+		},
+		{
+			name: "Substract a < b returns negative",
+			a:    3,
+			b:    5,
+			want: -2,
+		},
+		{
+			name: "Ensure decimals",
+			a:    6,
+			b:    2.225,
+			want: 3.77500,
+		},
+		{
+			name: "Substract a and b negatives which returns a positive",
+			a:    -3,
+			b:    -5,
+			want: 2,
+		},
+		{
+			name:  "Substract five numbers",
+			a:     20,
+			b:     5,
+			extra: []float64{4, 3, 7},
+			want:  1,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Subtract(tC.a, tC.b, tC.extra...)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		a     float64
+		b     float64
+		extra []float64
+	}{
+		{
+			name: "Multiply two positive numbers returns positive",
+			a:    3,
+			b:    2,
+			want: 6,
+		},
+		{
+			name: "Multiply positive and negative should return negative",
+			a:    3,
+			b:    -2,
+			want: -6,
+		},
+		{
+			name: "Multiply two negative numbers returns positive",
+			a:    -3,
+			b:    -2,
+			want: 6,
+		},
+		{
+			name: "Multiply fractions return fraction",
+			a:    3.2,
+			b:    2.501,
+			want: 8.0032,
+		},
+		{
+			name: "Multiply by zero return zero",
+			a:    1943,
+			b:    0,
+			want: 0,
+		},
+		{
+			name:  "Multiply five numbers",
+			a:     2,
+			b:     2,
+			extra: []float64{5, 3, 2},
+			want:  120,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Multiply(tC.a, tC.b, tC.extra...)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+// It only consider errExpected variable if an error occours.
+func TestDivide(t *testing.T) {
+	testCases := []struct {
+		name        string
+		errExpected bool
+		want        float64
+		a           float64
+		b           float64
+		extra       []float64
+	}{
+		{
+			name:        "Divide two positive integers",
+			errExpected: false,
+			want:        2,
+			a:           6,
+			b:           3,
+			extra:       nil,
+		},
+		{
+			name:        "Divide by zero",
+			errExpected: true,
+			want:        0,
+			a:           6,
+			b:           0,
+			extra:       nil,
+		},
+		{
+			name:        "Divide two negative integers",
+			errExpected: false,
+			want:        3,
+			a:           -6,
+			b:           -2,
+			extra:       nil,
+		},
+		{
+			name:        "Divide five numbers",
+			errExpected: false,
+			want:        1,
+			a:           60,
+			b:           2,
+			extra:       []float64{3, 5, 2},
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got, err := calculator.Divide(tC.a, tC.b, tC.extra)
+			if err != nil && tC.errExpected == false {
```

Comment:
There's a slightly subtle issue here, which I go into in more detail in the book. Suppose we have a test case which should not error, but does. Then this `if` statement will pick it up and fail the test. But suppose we have a test case which _should_ error (such as `Divide by zero`), but in fact doesn't? How can we catch that case, too?